### ISSUE TITLE
[PHP] Add option to disable ASP tag, script tag and short open tag

### ISSUE THREAD

**Issue by zufuliu** [ScintillaOrg/lexilla#187]
See https://wiki.php.net/rfc/remove_alternative_php_tags and https://wiki.php.net/rfc/deprecate_php_short_tags, they were removed from current PHP, only standard tag `<?php ?>` and short echo tag `<?= ?>` are supported.

for PHP code `<?=1//%>+2?>`, with ASP tag enabled line comment terminated at `%>`, else it terminated at `?>`.
pseudo code to fix this (inside `stateAllowsTermination()`):
```
case SCE_HPHP_COMMENTLINE:
	allowTermination = aspTagOn || ch == '?'; // ?>
	break;
```

**Comment by zufuliu** [comment]
Fix this could be delayed until someone has real problem with these removed tags?

**Comment by zufuliu** [comment]
> until someone has real problem with these removed tags

Found https://sourceforge.net/p/scintilla/bugs/905/

**Comment by zufuliu** [comment]
Patch added `lexer.php.asp.tag` property.

default on: [php-asp-tag-on.patch](https://github.com/ScintillaOrg/lexilla/files/12135929/php-asp-tag-on.patch)
default off: [php-asp-tag-off.patch](https://github.com/ScintillaOrg/lexilla/files/12135930/php-asp-tag-off.patch), this matches PHP's behavior.

No change (looks complex) for script tag and short open tag as they does not cause any problem in past.
https://github.com/ScintillaOrg/lexilla/blob/5ffca5619467ae8beae9210b964eb2acdd431b29/lexers/LexHTML.cxx#L110-L111
https://github.com/ScintillaOrg/lexilla/blob/5ffca5619467ae8beae9210b964eb2acdd431b29/lexers/LexHTML.cxx#L1383-L1394

**Comment by zufuliu** [comment]
[html-asp-tag.patch](https://github.com/ScintillaOrg/lexilla/files/12137396/html-asp-tag.patch)
Another patch, add `lexer.html.asp.tag` property (default on), so ASP tag can be turned off when using `SCLEX_HTML` (previous two patches can only turn off ASP tag with `SCLEX_PHPSCRIPT`).

**Comment by nyamatongwe** [comment]
`lexer.html.asp.tag` means applications would have to manage file types more by setting it on for ASP and off for PHP.

If `<?` may also be removed later then it may be worth experimenting with a simple implementation to define how any properties will interact. This is complicated by different server-side languages using similar tokens.

**Comment by zufuliu** [comment]
Fixed by issue #252.
